xm-commons-topic/src/main/java/com/icthh/xm/commons/topic/service/AutoCreateTopicService.java

```diff
@@ -0,0 +1,50 @@
+package com.icthh.xm.commons.topic.service;
+
+import com.icthh.xm.commons.topic.config.AutoCreateTopicConfiguration;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.kafka.clients.admin.*;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.autoconfigure.kafka.KafkaProperties;
+import org.springframework.stereotype.Component;
+
+import javax.annotation.PostConstruct;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+
+@Slf4j
+@Component
+@RequiredArgsConstructor
+@ConditionalOnProperty(value = "xm-topic.auto-create.enable", matchIfMissing = false, havingValue = "true")
```

Comment:
We try to use application.XXXX naming for feature flags grouping. 
@ConditionalOnProperty(value = "application.auto-system-queue-enabled", havingValue = "true")
@ConditionalOnProperty(value = "application.kafkaHealth.enabled", matchIfMissing = false)
please use same root

Reply:
@major13ua it also requires changes to classes such as ApplicationProperties in communication-app. This is normal?

Reply:
@amedvedchuk @sergeysenja1992 is there are common approach on this question?

Reply:
In case it requires to modify ApplicationProperties during commons update I would prefer to avoid it. 
Previously we have used property prefixes similar to commons like this: `@ConditionalOnProperty("xm-config.enabled")`.

Reply:
I propose start using 1 common root.
xm-commons.%name-of-commons%.your-property
This allows group property in 1 root, and then relates them to commons we are tuning by this property.

Reply:
> @major13ua it also requires changes to classes such as ApplicationProperties in communication-app. This is normal?

I propose start using 1 common root.
xm-commons.%name-of-commons%.your-property
This allows group property in 1 root, and then relates them to commons we are tuning by this property.